src/main/kotlin/net/maizegenetics/phgv2/cli/AgcCompress.kt

```diff
@@ -0,0 +1,228 @@
+
+        // check if ${dbPath}/assemblies.agc file exist
+        if (Files.exists(File("${dbPath}/assemblies.agc").toPath())) {
+            // if it exists, check if the fasta files in the fastaList/fastaFiles are already in the agc file
+            // Print a list of the duplicates, add the non-duplicates to the compressed file.
+            val duplicateList =
+                compareNewExistingSampleNames("${dbPath}/assemblies.agc", getCurrentSampleNames(fastaFiles),tempDir)
+            if (duplicateList.isNotEmpty()) {
+                myLogger.info("The following fasta files are already represented in the AGC compressed file and will not be loaded: ${duplicateList}")
+            }
+            //fastaFiles.forEach { sampleNames.add(it.split("/").last().split(".").first())
+            val listToLoad = fastaFiles.filter { !duplicateList.contains(it.split("/").last().split(".").first()) }
+            // Write the listToLoad to a file named tempDir/nonDuplicateFastaFiles.txt
+            val fileToLoad = File("${tempDir}/nonDuplicateFastaFiles.txt")
+            fileToLoad.writeText(listToLoad.joinToString("\n"))
+
+            if (listToLoad.isNotEmpty()) {
+                // Call method to load AGC files with the list of fasta files and load option
+                val success = loadAGCFiles(fileToLoad.toString(), "append", dbPath, refFasta,tempDir)
+            } else {
+                myLogger.info("No new fasta files to load -returning")
+            }
+        } else {
+            // call method to load AGC files with the list of fasta files and load option
+            val success = loadAGCFiles(fastaList, "create",dbPath,refFasta, tempDir)
+        }
+
+    }
+
+    fun loadAGCFiles(fastaFiles: String, loadOption: String, dbPath:String, refFasta:String, tempDir:String): Boolean {
+
+        val agcFile = "${dbPath}/assemblies.agc"
+        val agcFileOut = "${dbPath}/assemblies_tmp.agc"
+        val builder = ProcessBuilder()
+        var redirectOutput = tempDir + "/agc_create_output.log"
+        var redirectError = tempDir + "/agc_create_error.log"
+        when (loadOption) {
+            "create" -> {
+                builder.command("conda","run","-n","phgv2-conda","agc","create","-i",fastaFiles,"-o",agcFile,refFasta)
+            }
+            "append" -> {
+                builder.command("conda","run","-n","phgv2-conda","agc","append","-i",fastaFiles,agcFile,"-o",agcFileOut)
+                redirectOutput = tempDir + "/agc_append_output.log"
+                redirectError = tempDir + "/agc_append_error.log"
+            }
+            else -> {
+                println("Invalid load option: ${loadOption}")
+                return false
+            }
+        }
+
+        builder.redirectOutput( File(redirectOutput))
+        builder.redirectError( File(redirectError))
+
+        println("begin Command to create/append:" + builder.command().stream().collect(Collectors.joining(" ")))
```

Comment:
Use the logger and use a joinToString instead of .stream().collect()

Reply:
done